tests/TestCase/Http/ClientTest.php

```diff
@@ -898,4 +898,96 @@ public function testRedirectDifferentSubDomains()
         $this->assertInstanceOf(Response::class, $result);
         $this->assertSame($response, $result);
     }
+
+    /**
+     * Scheme is set when passed to client in string
+     */
+    public function testCreateScopedClientSetsProtocol()
+    {
+        $client = Client::createScopedClientFromUrl('https://example.co/');
+        $this->assertSame('https', $client->getConfig('scheme'));
+    }
+
+    /**
+     * Host is set when passed to client in string
+     */
+    public function testCreateScopedClientSetsHost()
+    {
+        $client = Client::createScopedClientFromUrl('https://example.co/');
+        $this->assertSame('example.co', $client->getConfig('host'));
+    }
+
+    /**
+     * Test exception is thrown when URL cannot be parsed
+     */
+    public function testCreateScopedClientThrowsInvalidExceptionWhenUrlCannotBeParsed()
+    {
+        try {
+            Client::createScopedClientFromUrl('htps://');
+        } catch (InvalidArgumentException $e) {
```

Comment:
You can use `expectException()` and `expectExceptionMessage()` instead of this try/catch.

Reply:
Corrected.